### PR TITLE
[CST-857] bug: participant show breaks when mentor has been removed by a sit

### PR DESCRIPTION
**Pull request by ltello**
### Context

- Ticket: [BUG - Participant show breaks when mentor has been removed ](https://dfedigital.atlassian.net/browse/CST-857)

This happens because there is some inconsistency in the models for this situations:

```
induction_record.mentor_profile.status #=> ‘withdrawn’
induction_record.mentor_profile.user.mentor_profile #=> nil  !!!
```

The  reason for that is: 

```
class TeacherProfile
  has_one :mentor_profile, -> { active_record }, class_name: "ParticipantProfile::Mentor"
end
```

So from an induction_record we can get to the mentor’s user associated with this record.

However if we go backwards, from that mentor user we can’t get to that user’s mentor profile or their associated induction_record.

### Changes proposed in this pull request
 - Make the view not to require to go backwards and receive @mentor_profile instead of @mentor to retrieve the  mentor_profile.id
 - Move to a service object all the logic for a SIT to remove a participant from a school cohort.
 - Reuse the extracted logic in the admin dashboard to remove participants exactly the same way as a SIT does.

### CODE REVIEW

File: db/migrate/20220725205758_dissociate_withdrawn_mentors_from_induction_records.rb [hunk 1]
```diff
@@ -0,0 +1,9 @@
+# frozen_string_literal: true
+
+class DissociateWithdrawnMentorsFromInductionRecords < ActiveRecord::Migration[6.1]
+  def change
+    InductionRecord.joins(:mentor_profile)
+                   .where(participant_profiles: { status: :withdrawn })
+                   .update_all(mentor_profile_id: nil)
+  end
+end
```
Comment (tonyheadford) [inline db/migrate/20220725205758_dissociate_withdrawn_mentors_from_induction_records.rb:9]: I don't think we need this?

**Review by tonyheadford — CHANGES_REQUESTED** [review]
Looks good except I don't think we want that migration?

File: db/migrate/20220725205758_dissociate_withdrawn_mentors_from_induction_records.rb (hunk 1 above)
Comment (ltello) [reply db/migrate/20220725205758_dissociate_withdrawn_mentors_from_induction_records.rb:9]: Ok, this was when I thought we were going to nullify all the induction records mentored by the mentor deleted. 
I will remove it now

**Review by tonyheadford — APPROVED** [review]

File: db/migrate/20220725205758_dissociate_withdrawn_mentors_from_induction_records.rb (hunk 1 above)
Comment (ltello) [reply db/migrate/20220725205758_dissociate_withdrawn_mentors_from_induction_records.rb]: Re-added the migration for current induction records still linked to withdrawn mentor profiles.

**Pull request merged**